kore/src/Kore/Step/AntiLeft.hs

```diff
@@ -0,0 +1,417 @@
+{-|
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+-}
+
+module Kore.Step.AntiLeft
+    ( AntiLeft (..)
+    , antiLeftPredicate
+    , forgetSimplified
+    , mapVariables
+    , parse
+    , substitute
+    , toTermLike
+    ) where
+
+import Prelude.Kore
+
+import Control.DeepSeq
+    ( NFData
+    )
+import Data.Foldable
+    ( fold
+    , toList
+    )
+import Data.List
+    ( foldl'
+    )
+import Data.Map.Strict
+    ( Map
+    )
+import qualified Data.Map.Strict as Map
+import Data.Set
+    ( Set
+    )
+import qualified Data.Set as Set
+import qualified Generics.SOP as SOP
+import qualified GHC.Generics as GHC
+
+import Debug
+    ( Debug
+    , Diff
+    )
+import Kore.Attribute.Pattern.FreeVariables
+    ( HasFreeVariables (freeVariables)
+    , bindVariables
+    )
+import qualified Kore.Attribute.Pattern.FreeVariables as FreeVariables
+    ( toNames
+    )
+import Kore.Internal.Predicate
+    ( Predicate
+    , makeAndPredicate
+    , makeCeilPredicate_
+    , makeMultipleExists
+    , makeMultipleOrPredicate
+    , makeOrPredicate
+    )
+import qualified Kore.Internal.Predicate as Predicate
+    ( forgetSimplified
+    , mapVariables
+    , substitute
+    , wrapPredicate
+    )
+import Kore.Internal.TermLike
+    ( pattern And_
+    , pattern ApplyAlias_
+    , pattern Bottom_
+    , pattern Exists_
+    , pattern Or_
+    , TermLike
+    , mkAnd
+    , mkElemVar
+    )
+import qualified Kore.Internal.TermLike as TermLike
+    ( forgetSimplified
+    , mapVariables
+    , substitute
+    )
+import Kore.Internal.Variable
+    ( InternalVariable
+    )
+import Kore.Step.Simplification.ExpandAlias
+    ( substituteInAlias
+    )
+import Kore.Syntax.Variable
+    ( AdjSomeVariableName
+    , ElementVariable
+    , SomeVariableName (SomeVariableNameElement)
+    , Variable (variableName)
+    , mapElementVariable
+    )
+import Kore.Variables.Fresh
+    ( FreshPartialOrd
+    , refreshElementVariable
+    )
+
+data AntiLeftLhs variable = AntiLeftLhs
```

Comment:
Please add documentation explaining what this type represents. I think it's the left-hand side of one rule used in the anti-left pattern? But, I don't know why the alias and its expansion are both present.

Reply:
I removed the alias, it made sense in an older version of this PR, but doesn't anymore.

I added some explanations, is it more clear now?